warehouse/ingest-scripts/src/main/resources/bin/util/buildIndexStats.sh

```diff
@@ -0,0 +1,365 @@
+#
+# Higher precision values translate to more buckets and thus higher memory utilization.
+# The number of buckets that are allocated is modeled by  the equation:
+#
+#        buckets = 2 ** n  => where n is the precision
+#
+# As the precision is increased, the error percentage decreases and is modeled by
+# the equation:
+#
+#        error percentage = 1.0 / SQRT(m)   => where m is the number of buckets
+#
+# Precision values should be set based upon memory usage versus error percentage. Several
+# tests indicate that inconsistent results can occur when the sparse and normal precision
+# values differ. The greater the difference, the larger the error percentage. While most
+# of the generated cardinalities fell within the projected error percentage, in the few
+# instances where it did it was off by very significat factor. Thus when selecting
+# sparse and normal precision values it is best to chose the same value for both.
+#
+#+++++++++++++++++++++++++++++++++++++++
+# Optional Debug Environment Variables
+# There are several options that will allow for analysis of the debug data:
+#   _UniqueCounts => Collects actual unique counts in addition to the HyperLogLogPlus
+#           generated cardinality for analysis purposes. The mapper will generate the
+#           the unique count, while the reducer is the best location to view all of the
+#           entries from multiple mappers. The test script hlParse.sh is provided to
+#           parse the lo file of the reducer. This process extracts the analysis data
+#           into a csv format, which can be loaded into a spreadsheet for analyis.
+#   _GCMapperLog => Enables GC logging for the mapper.
+#   _GCReducerLog => Enables GC logging for the reducer.
+#   _GCLog => Enables GC logging for both the mapper and reducer.
+#   _GCLogDir => Specifies the directory for the GC logs (overrides the deault value).
+#   _HeapDump => Enable heap dump for mapper and reducer.
+#========================================================================================
+
+if [[ `uname` == "Darwin" ]]; then
+        THIS_SCRIPT=`python -c 'import os,sys;print os.path.realpath(sys.argv[1])' $0`
+else
+        THIS_SCRIPT=`readlink -f $0`
+fi
+
+THIS_DIR="${THIS_SCRIPT%/*}"
+cd $THIS_DIR
+
+. ../ingest/ingest-env.sh
+
+declare -r _base=$(basename $0)
+declare -r _lockFile=${LOCK_FILE_DIR}/${_base}
+
+
+#===================================================================================
+# local functions
+function cleanup() {
+    declare -r _status=$1
+
+    rm -f "${_lockFile}"
+    trap 0 1 2 3 15
+    exit ${_status}
+}
+
+function setupLockFile() {
+    # check for current execution of this script
+    test -e ${_lockFile} && {
+        echo "ERROR: lock file exists => ${_lockFile}"
+        exit 1
+    }
+    touch "${_lockFile}"
+    trap "cleanup 1" 1 2 3 15
+    trap "cleanup 0" 0
+}
+
+#==================================
+# Sets JVM parameters for Mapper
+function setMapperJVMParameters() {
```

Comment:
Why are the memory values hardcoded to these specific values?

Reply:
A different script was used as the basis for this script, which had hardcoded values. I guess I could use the default but I still need to set the io sort buffer size.

Reply:
The memory value will be allow to be a set by an environmental variable.